pkg/controller/pxc/upgrade.go

```diff
@@ -38,16 +38,19 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 	// embed DB configuration hash
 	// TODO: code duplication with deploy function
 	configHash := r.getConfigHash(cr, sfs)
-	if currentSet.Spec.Template.Annotations == nil {
-		currentSet.Spec.Template.Annotations = make(map[string]string)
-	}
+	currentSet.Spec.Template.Annotations = make(map[string]string)
```

Comment:
Same here, what if user already have some annotations?

Reply:
Same here =)
Service annotations will be filled again, but if we remove some annotations from `cr`, annotations will be updated since we are not exist in new cr